### PR TITLE
Started on ChannelCommand... For real now

### PR DESCRIPTION
**Pull request by Addison118**
Here we are again :p

### CODE REVIEW

File: src/legionpe/theta/Session.php [hunk 1]
```diff
@@ -930,6 +930,12 @@ public function joinChannel($channel, $subLevel = self::CHANNEL_SUB_NORMAL){
 		new JoinChannelQuery($this->getMain(), $this->getUid(), $channel, $subLevel);
 		return true;
 	}
+    public function getCurrentChatState() {
+        return $this->currentChatState;
+    }
+    public function setCurrentChatState() {
```
Comment (PEMapModder) [inline src/legionpe/theta/Session.php:936]: Do you know what a setter is ... A setter is to set a property into the parameter passed into the function...

File: src/legionpe/theta/command/session/ChannelCommand.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        parent::__construct($plugin, "channel", "switch channels", "/channel <channel...>");
+    }
+
+    public function run(array $args, Session $session)
+    {
+        if (strtolower($args[0]) == "local") {
+            if($session->getCurrentChatState() === Session::CHANNEL_LOCAL) {
+                $session->getPlayer()->sendMessage(TextFormat::RED . "You are already in the local channel");
+            } else {
+                $session->joinChannel(Session::CHANNEL_LOCAL);
```
Comment (PEMapModder) [inline src/legionpe/theta/command/session/ChannelCommand.php]: Local and team aren't channels. They are special chat states that are handled in another way.

File: src/legionpe/theta/command/session/ChannelCommand.php [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+                $session->currentChatState = Session::CHANNEL_LOCAL;
+            }
+        }
+        if (strtolower($args[0]) == "team") {
+            if($session->getCurrentChatState() === Session::CHANNEL_TEAM) {
+                $session->getPlayer()->sendMessage(TextFormat::RED . "You are already in your team channel");
+            } else {
+                $session->joinChannel(Session::CHANNEL_TEAM);
+                $session->currentChatState = Session::CHANNEL_TEAM;
```
Comment (PEMapModder) [inline src/legionpe/theta/command/session/ChannelCommand.php:44]: It is a private property; you can't access it directly...

File: src/legionpe/theta/command/session/ChannelCommand.php [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+            }
+        }
```
Comment (PEMapModder) [inline src/legionpe/theta/command/session/ChannelCommand.php:46]: What about other real channels?

**Comment by Addison118** [comment]
why were you telling me to use $session->currentChatState = Session::CHANNEL_TEAM?

**Comment by Addison118** [comment]
Also, if local and team are not channels... why do they say CHANNEL_LOCAL and CHANNEL_TEAM?

**Pull request closed without merging**